packages/mobile/src/harmony-native/components/Button/FilterButton/FilterButton.tsx

```diff
@@ -0,0 +1,169 @@
+    width: spacing.unit4,
+    height: spacing.unit4
+  }
+
+  const smallStyles: ReactNativeStyle = {
+    paddingHorizontal: spacing.m,
+    paddingVertical: spacing.xs
+  }
+  const smallIconStyles: ReactNativeStyle = {
+    width: spacing.unit3,
+    height: spacing.unit3
+  }
+
+  // State Styles
+  const fillContainerStyles: ReactNativeStyle = {
+    backgroundColor: color.secondary.s400,
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.secondary.s400
+  }
+
+  const activeStyle: ReactNativeStyle =
+    variant !== 'fillContainer' || value === null
+      ? {
+          backgroundColor: color.background.surface2,
+          borderWidth: 1,
+          borderStyle: 'solid',
+          borderColor: color.border.strong
+        }
+      : {}
+
+  // Button Styles
+  const buttonStyles: ReactNativeStyle = {
+    backgroundColor: color.background.white,
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.border.strong,
+    borderRadius: cornerRadius.s,
+    color:
+      variant === 'fillContainer' && value !== null
+        ? color.static.white
+        : color.text.default,
+    gap: spacing.xs,
+    fontSize: typography.size.s,
+    fontWeight: '600',
+    lineHeight: typography.lineHeight.s,
+    opacity: disabled ? 0.6 : 1,
+
+    ...(size === 'small' ? smallStyles : defaultStyles),
+    ...(isOpen ? activeStyle : {}),
+    ...(variant === 'fillContainer' && value !== null
+      ? fillContainerStyles
+      : {})
+  }
+
+  const iconStyles = size === 'small' ? smallIconStyles : defaultIconStyles
+
+  useEffect(() => {
+    if (isOpen) {
+      onOpen?.()
+    }
+  }, [isOpen, onOpen])
+
+  const handlePress = useCallback(() => {
+    if (onPress) {
+      onPress()
+    } else {
+      if (variant === 'fillContainer' && value !== null) {
+        setValue(null)
+        setLabel(null)
+        // onChange?.(null)
+        onReset?.()
+      } else {
+        setIsOpen((isOpen: boolean) => !isOpen)
+      }
+    }
+  }, [onPress, onReset, setLabel, setValue, value, variant])
+
+  // const handleChange = useCallback(
+  //   (value: string, label: string) => {
+  //     setValue(value)
+  //     setLabel(label)
+  //     onChange?.(value)
+  //   },
+  //   [onChange, setValue, setLabel]
+  // )
+
+  const iconSize = size === 'small' ? 's' : 'm'
+  const textColor =
+    value !== null && variant === 'fillContainer' ? 'staticWhite' : 'default'
+
+  return (
+    <BaseButton
+      style={buttonStyles}
+      styles={{ icon: iconStyles }}
+      innerProps={{
+        icon: {
+          color: 'staticWhite',
+          size: iconSize
+        }
+      }}
+      onPress={handlePress}
+      iconRight={
+        variant === 'fillContainer' && value !== null ? IconCloseAlt : iconRight
+      }
+      disabled={disabled}
+      aria-haspopup='listbox'
+      aria-expanded={isOpen}
+    >
+      <Text color={textColor}>{label}</Text>
+      {/* {children?.({ */}
```

Comment:
just wondering what youre thinking here? my guess is that we actually wont do children for filter button, itll just open up a screen? in that case we can build that into the api by saying: `filterScreen` as a prop which may be a "to" object that gets navigated to when you click?

Reply:
yea i agree. i think it might not be needed on mobile bc the button will usually open up another screen or drawer or something. I think that this implementation using just the onPress allows for the toggle style filters that don't open another screen. i can remove the comments 👍